test/p2p-test.js

```diff
@@ -50,15 +52,92 @@ describe('Compact Blocks', function() {
   });
 
   const nodePackets = [];
+  const nodePacketsAll = [];
 
   node.pool.on('packet', (packet) => {
     nodePackets.push(packet);
+    if (packet.type === packets.types.CFILTER || packet.type === packets.types.CFHEADERS || packet.type === packets.types.CFCHECKPT) {
+      nodePacketsAll.push(packet);
+    }
   });
 
   beforeEach(() => {
     nodePackets.length = 0;
   });
 
+  describe('BIP157', () => {
+    let block;
+
+    before(async () => {
+      for (let i = 0;i < 996;i++) {
+        block = await node2.miner.mineBlock();
+        await node2.chain.add(block);
+      }
+      await forValue(node.chain, 'height', node2.chain.height);
+    });
+
+    it('CFilters', async () => {
+      const pkt = new packets.GetCFiltersPacket(
+        0,
+        0,
+        block.hash()
+      );
+      const peer = node.pool.peers.head();
+
+      peer.send(pkt);
+      await new Promise(r => setTimeout(r, 20000));
+      let count = 0;
+      for (const p of nodePacketsAll) {
+        if (p.type === packets.types.CFILTER) {
+          // console.log(p);
+          count++;
+        }
+      }
+      // console.log(count);
+      assert(count === 999, '999 CFilters should be received');
+    });
+
+    it('CFHeaders', async () => {
+      const pkt = new packets.GetCFHeadersPacket(
+        0,
+        0,
+        block.hash()
+      );
+      const peer = node.pool.peers.head();
+
+      peer.send(pkt);
+      await new Promise(r => setTimeout(r, 100));
+      // for (const p of nodePacketsAll) {
```

Comment:
TODO: Cleanup these comments

Reply:
These are for Matthew to test the code while reviewing the PR.